packages/block-editor/src/components/use-block-sync/block-id-map.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { v4 as uuid } from 'uuid';
+
+/**
+ * Internal dependencies
+ */
+import { DIRECTION_IN, DIRECTION_OUT } from './constants';
+
+const BLOCK_ID_MAP = {};
```

Comment:
It looks like BLOCK_ID_MAP should only be accessible to internal methods. I think we could more clearly indicate that it's private by encapsulating the data structure in a class, or a closure. At the very least, we could add an underscore to the variable name `_BLOCK_ID_MAP`.

```javascript
// It looks like the only way to create private fields in ES6 is by scoping everything within the constructor
class BlockIdMapper {
  constructor {
    const BLOCK_ID_MAP = {};
    
    this.initBlockMap = () => {}
    this.mapBlockId = () => {}
    this.mapBlocks = () => {}
  }
}

```

Reply:
I'm unsure about classes because I don't like using singletons, and I fear we would have to use one to share the object globally 🤔 We could also easily refactor this data structure so that we have one class per instance, and then store it in `useRef` for access in the hook 🤔 

but I totally agree that we can be more clear about making it private!

Reply:
Responded below, taking the class idea a bit further.

Reply:
> I'm unsure about classes because I don't like using singletons

Catching up with the pitfalls of singletons. If anyone else is curious, there are good resources [here](https://stackoverflow.com/questions/1020312/are-singletons-really-that-bad) and [here](https://www.davidtanzer.net/david's%20blog/2016/03/14/6-reasons-why-you-should-avoid-singletons.html)

> I fear we would have to use one to share the object globally

Gotcha! That makes sense.

Reply:
And when I say I don't like using singletons, it's at least 70% personal preference :D (Though there are practical concerns like in the articles you mention.) 

I mostly don't like the added verbosity of something like `BlockManager.getInstance()` all over the place